src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -942,6 +932,40 @@ public boolean equals(Object o) {
     return false;
   }
 
+  private boolean equals(RunContainer runContainer) {
+    if (runContainer.nbrruns != this.nbrruns) {
+      return false;
+    }
+    for (int i = 0; i < nbrruns; ++i) {
+      if (getValue(i) != runContainer.getValue(i)) {
+        return false;
+      }
+      if (getLength(i) != runContainer.getLength(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean equals(ArrayContainer arrayContainer) {
+    int pos = 0;
+    for (short i = 0; i < nbrruns; ++i) {
+      short runStart = getValue(i);
+      short length = getLength(i);
+      if (pos + length >= arrayContainer.getCardinality()) {
+        return false;
```

Comment:
In a library which models sets of integers, there are dozens of simple invariants (the equals method should be reflexive, if bitmaps are equal, cardinalities should be equal, A and not A and not B should equals A, the list goes on...). Basically there are plenty of hypotheses from which to generate test cases faster than you could possibly think of them...